mkctr.go

```diff
@@ -456,7 +470,10 @@ func layerFromFiles(logf logf, files map[string]string) (v1.Layer, error) {
 		return nil, err
 	}
 
-	return tarball.LayerFromReader(buf, tarball.WithCompressedCaching)
+	binaryLayerBytes := buf.Bytes()
+	return tarball.LayerFromOpener(func() (io.ReadCloser, error) {
+		return io.NopCloser(bytes.NewBuffer(binaryLayerBytes)), nil
+	}, tarball.WithCompressedCaching, tarball.WithMediaType(layerMediaType))
```

Comment:
This commit is largely pulled from ko https://github.com/ko-build/ko/blob/v0.15.1/pkg/build/gobuild.go#L806-L847

Reply:
The deprecation docs say
> Deprecated: Use LayerFromOpener or stream.NewLayer instead, if possible. 

should we instead use [stream.NewLayer](https://pkg.go.dev/github.com/google/go-containerregistry@v0.17.0/pkg/v1/stream#NewLayer)?


This is fine too, just curious and probably worth a comment.

Reply:
I took a look at the `stream.NewLayer` - it does not work for us out of the box because we actually need to call `image.Config()` to modify the entry command before the image is written - `stream.NewLayer` does not allow us to do that https://github.com/google/go-containerregistry/tree/main/pkg/v1/stream#caveats
Looking at how others do it, I see that they write the layer to upstream immediately after creating it https://github.com/kptdev/kpt/blob/main/pkg/wasm/client.go#L90. We _could_ also do it that way, but I am not sure if there would be any benefits to that- it seems to me that writing to the upstream repository once and as late as possible is actually preferable to avoid i.e uploading something incomplete when we later error etc?

Reply:
If no-one disagrees I will just add a comment as to why we don't use `stream.NewLayer`

Reply:
sgtm, thanks for the explanation!